proxy/src/test/java/com/wavefront/agent/PushAgentTest.java

```diff
@@ -173,6 +186,98 @@ public void testWavefrontUnifiedPortHandlerPlaintextUncompressed() throws Except
+    proxy.initSslContext();
+    proxy.startGraphiteListener(proxy.proxyConfig.getPushListenerPorts(), mockHandlerFactory, null);
+    waitUntilListenerIsOnline(port);
+    reset(mockPointHandler);
+    mockPointHandler.report(ReportPoint.newBuilder().setTable("dummy").
+            setMetric("metric.test").setHost("test1").setTimestamp(startTime * 1000).setValue(0.0d).build());
+    expectLastCall();
+    mockPointHandler.report(ReportPoint.newBuilder().setTable("dummy").
+            setMetric("metric.test").setHost("test2").setTimestamp((startTime + 1) * 1000).setValue(1.0d).build());
+    expectLastCall();
+    replay(mockPointHandler);
+
+    // try plaintext over tcp first
+    Socket socket = sslSocketFactory.createSocket("localhost", port);
+    BufferedOutputStream stream = new BufferedOutputStream(socket.getOutputStream());
+    String payloadStr = "metric.test 0 " + startTime + " source=test1\n" +
+            "metric.test 1 " + (startTime + 1) + " source=test2\n";
+    stream.write(payloadStr.getBytes());
+    stream.flush();
+    socket.close();
+    verifyWithTimeout(500, mockPointHandler);
+  }
+
+  @Test
+  public void testWavefrontUnifiedPortHandlerPlaintextUncompressedWithSSLAndPort() throws Exception {
+    port = findAvailablePort(2888);
+    proxy.proxyConfig.pushListenerPorts = String.valueOf(port);
+    proxy.proxyConfig.enableTLS = true;
+    proxy.proxyConfig.privateCertPath = getClass().getClassLoader().getResource("demo.cert").getPath();
+    proxy.proxyConfig.privateKeyPath = getClass().getClassLoader().getResource("demo.key").getPath();
+    proxy.proxyConfig.tlsPorts = "1,23 , 4,   , " + port +"  ,6";
+    proxy.initSslContext();
+    proxy.startGraphiteListener(proxy.proxyConfig.getPushListenerPorts(), mockHandlerFactory, null);
+    waitUntilListenerIsOnline(port);
+    reset(mockPointHandler);
+    mockPointHandler.report(ReportPoint.newBuilder().setTable("dummy").
+            setMetric("metric.test").setHost("test1").setTimestamp(startTime * 1000).setValue(0.0d).build());
+    expectLastCall();
+    mockPointHandler.report(ReportPoint.newBuilder().setTable("dummy").
+            setMetric("metric.test").setHost("test2").setTimestamp((startTime + 1) * 1000).setValue(1.0d).build());
+    expectLastCall();
+    replay(mockPointHandler);
+
+    // try plaintext over tcp first
+    Socket socket = sslSocketFactory.createSocket("localhost", port);
+    BufferedOutputStream stream = new BufferedOutputStream(socket.getOutputStream());
+    String payloadStr = "metric.test 0 " + startTime + " source=test1\n" +
+            "metric.test 1 " + (startTime + 1) + " source=test2\n";
+    stream.write(payloadStr.getBytes());
+    stream.flush();
+    socket.close();
+    verifyWithTimeout(500, mockPointHandler);
+  }
+
+  @Test
+  public void testWavefrontUnifiedPortHandlerPlaintextUncompressedWithSSLAndDifferentPort() throws Exception {
```

Comment:
I think it would be better to combine this test and a previous one into one test - have two ports in pushListenerPorts, start two listeners, one of them secured, another is not; and confirm that both work.